src/implicit_function.jl

```diff
@@ -24,103 +25,18 @@ end
 """
     ImplicitFunction(forward, conditions)
 
-Construct an [`ImplicitFunction{F,C,L}`](@ref) with `Krylov.gmres` as the default linear solver.
+Construct an [`ImplicitFunction{F,C,L}`](@ref) with [`Krylov.gmres`](https://juliasmoothoptimizers.github.io/Krylov.jl/stable/solvers/unsymmetric/#GMRES) as the default linear solver.
 """
-function ImplicitFunction(forward::F, conditions::C) where {F,C}
+function ImplicitFunction(forward, conditions)
     return ImplicitFunction(forward, conditions, gmres)
 end
 
-struct SolverFailureException{S} <: Exception
-    msg::String
-    stats::S
-end
-
-function Base.show(io::IO, sfe::SolverFailureException)
-    return println(io, "SolverFailureException: $(sfe.msg) \n Solver stats: $(sfe.stats)")
-end
-
 """
     implicit(x[; kwargs...])
 
 Make [`ImplicitFunction{F,C,L}`](@ref) callable by applying `implicit.forward`.
 """
-(implicit::ImplicitFunction)(x; kwargs...) = implicit.forward(x; kwargs...)
-
-"""
-    frule(rc, (_, dx), implicit, x[; kwargs...])
-
-Custom forward rule for [`ImplicitFunction{F,C,L}`](@ref).
-
-We compute the Jacobian-vector product `Jv` by solving `Au = -Bv` and setting `Jv = u`.
-Keyword arguments are given to both `implicit.forward` and `implicit.conditions`.
-"""
-function ChainRulesCore.frule(
```

Comment:
Why not keep this?

Reply:
Mostly because
- it's only useful for Diffractor at the moment
- I cannot rewrite it using AbstractDifferentiation because of https://github.com/JuliaDiff/AbstractDifferentiation.jl/issues/80